### PR TITLE
Repo builder for rpc-extras

### PR DESCRIPTION
**Pull request by d34dh0r53**
This patch adds the capability for rpc-extras to build wheels and
store them on the local repo server. The repo-build play scans
through all of the YAML files in the tree and creates wheels for
any packages or git sources it finds and places the wheels in the
rpc-extras tree of the repo server. The repo-pip-setup play adds
the rpc-extras tree to the pip configuration of all hosts. Due to
the way repo-build scans empty YAML files are not allowed, hence
the removal of the haproxy main.yml file.

### CODE REVIEW

**Comment by odyssey4me** [comment]
This looks good - I'd like to do a test run before merging it.

**Comment by claco** [comment]
:+1:

**Comment by odyssey4me** [comment]
To handle the edge case of the repo servers not mirroring immediately, can we perhaps do a restart notification for lsyncd in the repo-build play?

File: playbooks/repo-pip-setup.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  max_fail_percentage: 20
+  gather_facts: true
+  user: root
+  roles:
+    - pip_lock_down
+  post_tasks:
+    - name: Add new pip link
+      lineinfile:
+        dest: "{{ ansible_env.HOME }}/.pip/links.d/rpc_release.link"
+        line: "{{ openstack_repo_url }}/rpc-releases/{{ rpc_release }}"
```
Comment (odyssey4me) [inline playbooks/repo-pip-setup.yml]: Small detail for consistency, but not functionally required - there should be a '/' at the end of this line just before the quotes

**Comment by odyssey4me** [comment]
:+1: I've tested this and it's working.

**Comment by stevelle** [comment]
:+1: works for me
